epipearl/epipearl.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+
+"""
+epipearl client
+---------------
+defines the main api client class
+"""
+
+from . import __version__
+
+import os
+import sys
+import platform
+import requests
+
+from requests.auth import HTTPBasicAuth
+from urlparse import urljoin
+
+from endpoints import *
+
+__all__ = ['Epipearl', 'EpipearlError']
+
+_default_timeout = 5
+
+
+class EpipearlError( Exception ):
+    pass
+
+
+def handle_http_exceptions(callbacks=None):
+    if callbacks is None:
+        callbacks = {}
+    def wrapper(f):
+        def newfunc(*args, **kwargs):
+            try:
+                return f(*args, **kwargs)
+            except requests.HTTPError, e:
+                resp = e.response
+                req = e.request
+                if resp.status_code in callbacks.keys():
+                    callbacks[resp.status_code](e)
+                else:
+                    raise
+            except Exception, e:
+                raise
+        return newfunc
+    return wrapper
+
+
+def default_useragent():
```

Comment:
this was probably overkill in pyhorn. not sure why I didn't just use some combination of "Pyhorn" + `__version__`.